pkg/mesh/routes.go

```diff
@@ -232,6 +254,13 @@ func (t *Topology) Rules(cni bool) []iptables.Rule {
 		for _, aip := range s.allowedIPs {
 			rules = append(rules, iptables.NewRule(iptables.GetProtocol(len(aip.IP)), "nat", "KILO-NAT", "-d", aip.String(), "-m", "comment", "--comment", "Kilo: do not NAT packets destined for known IPs", "-j", "RETURN"))
 		}
+		// Make sure packages to allowed location IPs go through the KILO-NAT chain, so they can be MASQUERADEd,
+		// Otherwise packages to these destinations will reach the destitnation, but never find their way back.
+		for _, alip := range s.allowedLocationIPs {
+			rules = append(rules,
+				iptables.NewRule(iptables.GetProtocol(len(alip.IP)), "nat", "POSTROUTING", "-d", alip.String(), "-m", "comment", "--comment", "Kilo: jump to NAT chain", "-j", "KILO-NAT"),
```

Comment:
I think that in the worst case, this rule will cause packets to be NATed 3 times:
1. on the source machine
2. on the leader node of the source location and
3. on the leader node of the allowed IP location
To prevent this from happening, we should make sure that we only jump to the KILO-NAT chain for `allowedLocationIP`s that belong to the same segment as this kg instance. This would ensure that packets only get NATed once when they are about to be emitted by the node in the same LAN as the allowed IP. WDYT?

Reply:
Yes, so we need to pass more parameters to the `rules` function.

Reply:
I think this func should have all of the information it needs since it operates on the topology, right?